var/spack/repos/builtin/packages/eigen/package.py

```diff
@@ -30,6 +30,9 @@ class Eigen(CMakePackage):
     version('3.2.6', sha256='e097b8dcc5ad30d40af4ad72d7052e3f78639469baf83cffaadc045459cda21f')
     version('3.2.5', sha256='8068bd528a2ff3885eb55225c27237cf5cda834355599f05c2c85345db8338b4')
 
+    # there is a bug that provokes bad parsing of nvhpc version
+    patch('0001-Fix-parsing-of-version-for-nvhpc.patch', when='@3.2.6:3.3.9')
```

Comment:
I think we can download the patch and delete it from the PR:
```suggestion
    patch(
        'https://gitlab.com/libeigen/eigen/-/commit/001a57519a7aa909d3bf0cd8c6ec8a9cd19d9c70.patch',
        sha256='3bad83512dc24540bad0250379e0061fe360ce370a10bfeb4cc0dbdf138825f9', 
        when='@3.2.6:3.3.9'
    )
```